### PR TITLE
jgrss/nc filename

### PR DESCRIPTION
**Pull request by jgrss**
## What is this PR changing?
Adds user args to specify band names for co-registration.

### CODE REVIEW

File: src/geowombat/core/api.py [hunk 1]
```diff
@@ -516,42 +519,52 @@ def __init__(
...
-                        logger.exception('  The chunks should be a dictionary.')
+                    if "chunks" in kwargs and not isinstance(kwargs["chunks"], dict):
+                        logger.exception("  The chunks should be a dictionary.")
                         raise TypeError
 
                     with xr.open_dataset(filename, **kwargs) as src:
-                        self.data = src.to_array(dim='band')
+                        self.data = src.to_array(dim="band")
+                    # Ensure the filename attribute gets updated as the NetCDF file
+                    self.data = self.data.assign_attrs(**{"filename": str(filename)})
```
Comment (jgrss) [inline src/geowombat/core/api.py:533]: This is a fix to ensure the NetCDF filename replaces the filename attribute stored in the NetCDF file.

File: src/geowombat/core/api.py [hunk 2]
```diff
@@ -516,42 +519,52 @@ def __init__(
...
 
+        self.data = self.data.assign_attrs(
+            {
+                "data_are_separate": int(self.__data_are_separate),
+                "data_are_stacked": int(self.__data_are_stacked),
+            }
+        )
         if persist_filenames:
-            self.data.attrs['filenames'] = self.__filenames
+            self.data = self.data.assign_attrs(**{"filenames": self.__filenames})
```
Comment (jgrss) [inline src/geowombat/core/api.py:560]: Improved attribute setting

File: src/geowombat/core/sops.py [hunk 3]
```diff
@@ -1079,22 +1192,21 @@ def subset(
...
-        ds_sub.attrs['transform'] = dst_transform
+        ds_sub.attrs["transform"] = dst_transform
 
         return ds_sub
 
     @staticmethod
     def coregister(
         target,
         reference,
+        band_names_reference: T.Optional[T.Sequence[str]] = None,
```
Comment (jgrss) [inline src/geowombat/core/sops.py:1206]: New optional arguments

File: src/geowombat/core/sops.py [hunk 4]
```diff
@@ -1157,65 +1273,93 @@ def coregister(
                 raise KeyError
 
         with tempfile.TemporaryDirectory() as tmp:
-            ref_path = Path(tmp) / '_reference.tif'
-            tar_path = Path(tmp) / '_target.tif'
-            with gw_.open(reference.filename, chunks={'band': -1, 'y': 256, 'x': 256}) as ref_src:
+            ref_path = Path(tmp) / "_reference.tif"
+            tar_path = Path(tmp) / "_target.tif"
+            ref_kwargs = (
```
Comment (jgrss) [inline src/geowombat/core/sops.py:1278]: Use `h5netcdf` engine if the file is NetCDF.

**Comment by jgrss** [comment]
@mmann1123 I finally registered `Black` so there are _many_ changes. However, I commented on the relevant changes to look at.

File: src/geowombat/core/sops.py (hunk 4 above)
Comment (jgrss) [reply src/geowombat/core/sops.py:1278]: My environment throws many errors if the this NetCDF engine is not used.

File: src/geowombat/core/sops.py (hunk 3 above)
Comment (mmann1123) [reply src/geowombat/core/sops.py:1200]: good idea

File: src/geowombat/core/sops.py (hunk 4 above)
Comment (mmann1123) [reply src/geowombat/core/sops.py:1272]: yeah netcdfs are amazing but fickle

**Pull request merged**